go/arrow/datatype_encoded.go

```diff
@@ -22,14 +22,16 @@ type EncodedType interface {
 }
 
 // RunEndEncodedType is the datatype to represent a run-end encoded
-// array of data.
+// array of data. ValueNullable defaults to true, but can be set false
+// if this should represent a type with a non-nullable value field.
 type RunEndEncodedType struct {
-	ends DataType
-	enc  DataType
+	ends          DataType
+	enc           DataType
+	ValueNullable bool
```

Comment:
Is it deliberate that the terminology hesitates between "enc", "encoded" and "value(s)" for the child values type/array?

Reply:
that's a very good point. I should make that more consistent